README.md

```diff
@@ -272,6 +272,117 @@ The fact that the memory is not circular obligates to take into account border c
 
 * If the transfer starts in an allowed memory address but tries to surpass the range during the transaction, it will only modify the bytes in the allowed range and discard the rest. The end of the transfer is taken as the end of the memory.
 
+## UART protocol
+
+### UART features
+
+This section describes how to get you started with the `SupernovaController` focusing on the UART protocol.
+
+* The supported features are: 
+    * Bus initialization.
+    * Setting of bus parameters such as baudrate, hardware handshake, parity, data size and stop bit.
+    * UART TX/RX transactions of up to 1024 bytes.
+
+### Basic UART Communication
+
+#### Generic operations
+
+1. ***Initializing the Supernova Device:***
+
+   Imports and initializes the `SupernovaDevice`. Optionally, specifies the USB HID path if multiple devices are connected:
+
+   ```python
+   from supernovacontroller.sequential import SupernovaDevice
+
+   device = SupernovaDevice()
+   # Optionally specify the USB HID path
+   device.open(usb_address='your_usb_hid_path')
+   ```
+
+   Call `open()` without parameters if you don't need to specify a particular device.
+
+2. ***Creating a UART controller Interface:***
+
+   Creates a UART controller interface:
+
+   ```python
+   uart = device.create_interface("uart")
+   ``` 
+
+3. ***Closing the Device:***
+
+   Closes the device when done:
+
+   ```python
+   device.close()
+   ```
+
+### Operations intended for the Supernova UART peripheral
+
+1. ***Setting Bus Voltage:***
+
+   Sets the bus voltage (in mV) for the UART bus. This step is required before initializing the bus:
+
+   ```python
+   success, response = uart.set_bus_voltage(3300)
+   ```
+
+2. ***Initializing the Supernova UART peripheral:***
+
+    Initializes the Supernova in UART peripheral:
+
+    ```python
+   success, response = uart.init_bus()
+   ```
+    Without any parameters, the UART peripheral initializes with the default values for baudrate (9600bps), parity (no parity), data size (8 bit), stop bit (one stop bit) and hardware handshake (no hardware handshake). Optionally, it is possible to set any of these parameters by specifying in the init_bus function:
+
+    ```python
+   success, response = uart.init_bus(baudrate=UartControllerBaudRate.UART_BAUD_115200, parity=UartControllerParity.UART_EVEN_PARITY)
+   ```
+3. ***Modifying the UART peripheral parameters***
+
+    It is also possible to configure/set any parameter after initialization (baudrate, parity, data size, stop bit and hardware handshake):
+
+    ```python
+   success, response = uart.set_parameters(stop_bit = UartControllerStopBit.UART_TWO_STOP_BIT, baudrate = UartControllerBaudRate.UART_BAUD_56000)
+   ```
+
+   If parameters are provided, it configures the parameters; otherwise, it retains the current settings.
```

Comment:
The parameters provided are configured. The ones that aren't retain their current setting.*